### ISSUE TITLE
Fix SkillOpt feedback scoping so old review feedback trains candidates without contaminating baseline/evaluator scoring

### ISSUE THREAD

**Issue by jerryfane** [gitmoot/gitmoot#162]
## Summary

The current SkillOpt feedback flow mixes three different uses of imported human review feedback:

1. optimizer training signal,
2. target prompt instruction,
3. evaluator stop-readiness label.

That caused the SmithyX run to behave incorrectly:

- The baseline was not only the previous skill. It was effectively `previous skill + old review feedback injected into the target task`, so it produced a fresh improved answer and scored artificially high.
- The candidate was judged using old review labels like `quality: poor`, `continue_mode: refine`, and `promote: no` as if those labels described the new candidate output.
- Because the old feedback was scoped to `baseline_review_outputs`, those labels should describe the previous reviewed outputs, not future candidate outputs.

The fix is to make feedback scope explicit and keep each component's input clean.

## Problem Observed

In `smithyx-one-shot-voice-trial-002-review-001`, the imported feedback was about old reviewed options:

```yaml
feedback_target: baseline_review_outputs
reviewed_skill_version: smithyx-one-shot-voice-writer@v1
quality: poor
continue_mode: refine
promote: no
```

During selection eval, baseline received the same feedback as live target instructions and generated:

```text
why does a spawned worker get the same wallet as the chat
```

That scored high:

```yaml
hard: 1.0
soft: 0.86
mixed_gate: 0.93
```

The first candidate generated:

```text
did each subagent get its own budget or just the user's quota
```

The evaluator mostly liked it, but because it found a minor unresolved candidate-specific concern and the old feedback had `quality: poor`, the old stop-readiness cap was applied:

```yaml
raw_dimensions:
  artifact_validity: 1.0
  human_feedback_resolution: 0.88
  task_completeness: 0.92
quality_status: failed
soft: 0.45
selection_gate_hard: 0
mixed_gate: 0.225
```

That is wrong because the `poor/refine/no` labels were about old reviewed outputs, not the candidate.

### 1. Baseline should be previous skill only

Baseline target run should use:

```text
previous_skill + task_input
```

It should not use old review feedback as repair instructions.

### 2. Candidate should be candidate skill only

Candidate target run should use:

```text
candidate_skill + same_task_input
```

The candidate skill should have internalized the feedback through optimizer updates. The target should not receive the old review feedback as a live instruction unless a future explicit mode requests that.

### 3. Optimizer receives full feedback

The optimizer should receive all imported review data:

- rankings,
- reviewer notes,
- quality,
- continue mode,
- promote decision,
- issue URL,
- reviewed skill version,
- item ids.

This is where `quality: poor`, `continue_mode: refine`, and `promote: no` belong.

### 4. Evaluator receives feedback as rubric, not stop labels

Evaluator should receive old feedback as rubric requirements:

```yaml
rubric_requirements:
  - avoid obvious rephrasing
  - preserve question-like response-triggering move
  - add a sharper real point
  - remove filler words like "finally"
  - avoid AI-written phrasing
feedback_scope: baseline_review_outputs
```

It must not automatically cap candidate score because old reviewed outputs were poor.

### 5. Candidate scoring is candidate-specific

For old feedback scoped to previous outputs:

- Contract failure: `hard=0`.
- Severe unresolved candidate failure: low score, possibly hard fail.
- Minor unresolved issue: `hard=1`, lowered soft score.
- Mostly resolved: normal soft score.
- Better than baseline: accept.

A candidate with a minor issue should not become:

```yaml
hard_for_selection: 0
soft: 0.45
```

It should more likely be:

```yaml
hard_for_selection: 1
soft: 0.70-0.85
```

### A. Add explicit feedback scope helper

```python
def feedback_is_about_previous_outputs(event: dict) -> bool:
    return str(event.get("feedback_target") or "").strip().lower() in {
        "baseline_review_outputs",
        "previous_review_outputs",
    }
```

Use this instead of inferring from `quality`, `promote`, or `continue_mode` alone.

### B. Split feedback context into separate views

Create two derived contexts:

```python
optimizer_feedback_context
candidate_evaluation_rubric
```

- `optimizer_feedback_context`: full imported feedback.
- `candidate_evaluation_rubric`: requirements extracted from reviewer notes, without old stop labels acting as candidate caps.

### C. Add target feedback modes

Add explicit target/evaluator/optimizer modes to summaries and internal config:

```yaml
baseline_target_feedback_mode: none
candidate_target_feedback_mode: none
optimizer_feedback_mode: full
evaluator_feedback_mode: rubric
```

Default behavior for SkillOpt review feedback should be:

- baseline target: no old feedback repair prompt,
- candidate target: no old feedback repair prompt,
- optimizer: full feedback,
- evaluator: rubric context.

### D. Strip old review feedback from target task prompts during selection

When constructing target prompts for baseline/candidate selection eval, exclude ranked feedback events whose `feedback_target` is previous-output scoped.

The evaluator can still receive those events through rubric context.

### E. Update evaluator cap logic

Current problematic behavior:

```python
if quality == "poor":
    cap = 0.45
```

New behavior:

```python
if feedback_is_about_previous_outputs(event):
    # Do not apply old quality/promote/continue_mode labels as candidate stop caps.
    # Convert reviewer notes into rubric requirements instead.
    continue
```

Candidate cap/failure should come only from actual candidate-specific unresolved issues.

### F. Add diagnostics

Summaries should expose:

```yaml
feedback_target: baseline_review_outputs
reviewed_skill_version: smithyx-one-shot-voice-writer@v1
baseline_target_feedback_mode: none
candidate_target_feedback_mode: none
optimizer_feedback_mode: full
evaluator_feedback_mode: rubric
old_feedback_stop_cap_applied_to_candidate: false
baseline_feedback_leakage_prevented: true
```

## Flow

```mermaid
flowchart TD
  A[Human reviews old skill outputs] --> B[Store scoped feedback]
  B --> C[Optimizer receives full feedback]
  C --> D[Create candidate skill]

  E[Previous skill] --> F[Baseline target run]
  D --> G[Candidate target run]

  B --> H[Convert feedback to evaluator rubric]
  F --> I[Compare baseline vs candidate]
  G --> I
  H --> I

  I -->|candidate wins| J[Publish candidate for human decision]
  I -->|candidate loses| K[Retry optimizer with structured rejection]
```

## Acceptance Criteria

- Baseline selection target prompt does not include old review feedback as repair instructions.
- Candidate selection target prompt does not include old review feedback as live instructions.
- Optimizer context still includes all imported ranked feedback items.
- Evaluator receives old feedback as rubric context.
- Old `quality/promote/continue_mode` labels do not cap candidate score when `feedback_target=baseline_review_outputs`.
- Candidate with minor unresolved issue keeps `hard=1` and receives a lower but non-catastrophic soft score.
- Summary reports feedback modes and whether old feedback stop caps were applied.
- Regression test covers the SmithyX case where old feedback is `quality: poor`, `continue_mode: refine`, `promote: no`, but a candidate mostly resolves it.

## References

- OpenAI evaluation best practices: https://platform.openai.com/docs/guides/evaluation-best-practices
- OpenAI graders: https://platform.openai.com/docs/guides/graders
- OpenAI model optimization: https://platform.openai.com/docs/guides/model-optimization
- LangSmith experiment analysis: https://docs.langchain.com/langsmith/analyze-an-experiment
- Prompt Optimization with Human Feedback: https://openreview.net/pdf?id=344O51eTPN

**Comment by jerryfane** [comment]
Implemented the feedback-scope fix discussed in this issue.

Summary of changes:

- Added shared feedback-scope helpers for previous-output feedback targets, including list-valued targets such as `feedback_target: [baseline_review_outputs]`.
- Updated the Gitmoot dataloader so previous-output review feedback is no longer injected into target prompts. The target still receives task/artifacts, while optimizer/evaluator keep the full review context.
- Added item diagnostics:
  - `feedback_scopes`
  - `target_feedback_mode`
  - `evaluator_feedback_mode`
  - `optimizer_feedback_mode`
- Updated the generic judge and landing-page judge prompts so old review labels are treated as prior-output rubric context, not as current-candidate vetoes.
- Updated evaluator scoring so `quality/promote/continue_mode` stop caps only apply to live candidate feedback, not `baseline_review_outputs` feedback.
- Added candidate rubric context in judge prompts so old feedback still drives candidate evaluation.
- Fixed feedback-direct trainer context to include train/val reviewed feedback while excluding held-out test feedback.
- Added regressions for:
  - old review feedback not leaking into target prompts
  - list-valued feedback targets
  - old review labels not creating stop caps
  - train/val feedback-direct context coverage
  - held-out test feedback not leaking into optimizer context

Verification:

- `PYTHONPATH=. .venv/bin/pytest -q tests/test_gitmoot_package.py tests/test_gitmoot_dataloader.py tests/test_gitmoot_adapter.py tests/test_gate_fail_closed.py`
- Result: `215 passed`
- `codex-crazyj exec review --uncommitted`
- Final review result: no introduced bugs found.

One note: the local `codex-crazyj` review environment could not run global `pytest` and later hit a temp-dir limitation for its own targeted test command, but the repo-local `.venv` test command above passed cleanly.

**Comment by jerryfane** [comment]
Closing as completed. The feedback-scope fix has an implementation summary, verification, and review result posted above.
